plugins/com.python.pydev.analysis/src/com/python/pydev/analysis/scopeanalysis/AbstractScopeAnalyzerVisitor.java

```diff
@@ -326,7 +332,8 @@ protected void addToNamesToIgnore(SimpleNode node, boolean finishClassScope, boo
                 //found match in names to ignore...
 
                 if (finishClassScope && foundScopeType == Scope.SCOPE_TYPE_CLASS
-                        && scope.getCurrScopeId() < single.scopeFound.getScopeId()) {
+                        && scope.getCurrScopeId() < single.scopeFound.getScopeId()
+                        || (!futureAnnotationsImported && foundScopeType == Scope.SCOPE_TYPE_ANNOTATION)) {
```

Comment:
This if clause seems to be missing some parenthesis...

I think it should be something as the code below (please run all the tests afterwards):

```
(
   finishClassScope && scope.getCurrScopeId() < single.scopeFound.getScopeId() && (
        foundScopeType == Scope.SCOPE_TYPE_CLASS ||
        (!futureAnnotationsImported && foundScopeType == Scope.SCOPE_TYPE_ANNOTATION)
    )
)
```

Reply:
I ran all tests and this change seems to work fine. Already updated this handle in the last commit ;)